### PR TITLE
Update libraries for current version of Rust

### PR DESCRIPTION
**Pull request by JohnMeyerhoff**
I ran into an error about a reference being passed immutable as well as mutable and believe that in newer versions of rust that is not allowed anymore. I updated the two dependencies to their latest versions and updated the source code to work with the changes made in those libraries.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Learning on a foreign project
+I was interested in Rust and cloned a Project, that was not maintained. There was no documentation of
+what the program should do or if it ever did something. When I started, it was failing at Compilation.
```
Comment (rap2hpoutre) [inline README.md]: Not sure these lines are relevant for the project: it makes sense in your fork, but would not be clear in this repository. What is your opinion?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+For Compiling on linux, you might need to install additional packages. For me, the following helped:
+
+``sudo apt install libalsaplayer-dev libudev-dev libasound2-dev``
+___
+# Compiling and running the project
+Cargo will go a long way in helping you here. It should be a simple ```cargo run```. Tests (``cargo test``) wil also run, currently there are no tests implemented.
```
Comment (rap2hpoutre) [inline README.md]:
```suggestion
Cargo will go a long way in helping you here. It should be a simple ```cargo run```. Tests (``cargo test``) will also run, currently there are no tests implemented.
```

**Review by rap2hpoutre — COMMENTED** [review]
Thank you so much for your contribution! 🎉  You resurrected this project! Just 2 optionnal comments and it's good to go!

**Comment by rap2hpoutre** [comment]
Fixes #1

**Comment by JohnMeyerhoff** [comment]
Sounds great 👍

**Comment by TwoHorus** [comment]
Sounds great 👍

File: README.md (hunk 1 above)
Comment (JohnMeyerhoff) [reply README.md]: Thats Correct, it's not needed here :)

**Review by rap2hpoutre — APPROVED** [review]
🍾

**Comment by rap2hpoutre** [comment]
Thank you!!

**Pull request merged**